src/core/filesystem.cc

```diff
@@ -605,7 +605,7 @@ S3FileSystem::ParsePath(
 {
   // Get the bucket name and the object path. Return error if input is malformed
   re2::RE2 s3_regex(
-      "s3://([a-z]+):([0-9]+)/([0-9a-z.-]+)(((/[0-9a-z.-_]+)*)?)");
+      "s3://([0-9a-z-.]+):([0-9]+)/([0-9a-z.-]+)(((/[0-9a-zA-Z.-_]+)*)?)");
```

Comment:
Should we include capital characters for all parts that allows characters?

Reply:
Bucket names should not contain upper-case letters. 
From what I understand the convention is to keep the hostname in lower case as well. I could support capital letters in the place of a hostname.

Reply:
S3 doesn't allow bucket names to have capitals? Or just a convention? We need to allow whatever S3 supports.
You should allow capitals in the hostname. Name resolution is case-insensitive but caps are allowed.

Reply:
Already fixed to allow caps in hostname. The s3 bucket name can contain only lower-case characters, numbers, periods, and dashes. Its is a rule/requirement and not just a convention.